src/ca_biositing/pipeline/tests/test_new_analysis_etl.py

```
@@ -0,0 +1,227 @@
+from unittest.mock import patch, MagicMock
+
+# --- ICP ETL TEST ---
+@patch("ca_biositing.pipeline.etl.extract.icp.gsheet_to_df")
+@patch("ca_biositing.pipeline.etl.extract.icp.get_run_logger")
+@patch("ca_biositing.pipeline.etl.transform.analysis.icp_record.get_run_logger")
+@patch("ca_biositing.pipeline.etl.load.analysis.icp_record.get_run_logger")
+@patch("ca_biositing.pipeline.etl.load.analysis.icp_record.get_local_engine")
+@patch("ca_biositing.pipeline.etl.transform.analysis.icp_record.normalize_dataframes")
+def test_icp_etl_full(
+    mock_normalize,
+    mock_load_logger,
```

Ruff (0.15.1)

[warning] Unused function argument: `mock_load_logger`

(ARG001)